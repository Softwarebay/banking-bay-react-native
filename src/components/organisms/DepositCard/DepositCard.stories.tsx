import { action } from '@storybook/addon-actions';
import { number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { ViewStory } from 'utils/storybook/ViewStory';

import { DepositCard } from './DepositCard';

storiesOf('Organisms/DepositCard', module)
	.addDecorator((getStory) => <ViewStory>{getStory()}</ViewStory>)
	.add('Ongoing', () => (
		<DepositCard
			amount={number('amount', 300000)}
			currency={text('currency', 'USD')}
			date={text('date', 'Sep 1 - Mar 1, 2022')}
			isFinished={false}
			lastIncome={number('lastIncome', 2560)}
			rate={number('rate', 0.08, { range: true, min: 0, max: 1, step: 0.01 })}
			testID="test-id"
			onPressExtendButton={action('onPressExtendButton')}
			onPressTopUpButton={action('onPressTopUpButton')}
			onPressWithdrawalButton={action('onPressWithdrawalButton')}
		/>
	))
	.add('Finished', () => (
		<DepositCard
			amount={number('amount', 300000)}
			currency={text('currency', 'USD')}
			date={text('date', 'Sep 1 - Mar 1, 2022')}
			lastIncome={number('lastIncome', 2560)}
			rate={number('rate', 0.08, { range: true, min: 0, max: 1, step: 0.01 })}
			testID="test-id"
			isFinished
			onPressExtendButton={action('onPressExtendButton')}
			onPressTopUpButton={action('onPressTopUpButton')}
			onPressWithdrawalButton={action('onPressWithdrawalButton')}
		/>
	));
