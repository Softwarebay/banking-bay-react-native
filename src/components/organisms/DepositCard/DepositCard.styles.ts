import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export const styles = StyleSheet.create({
	container: {
		padding: 20,
	},
	currency: {
		fontFamily: 'MulishBold',
		marginLeft: 5,
	},
	extendButton: {
		backgroundColor: colors.coral,
		flex: 1,
		marginRight: 5,
	},
	floatStyle: {
		fontFamily: 'MulishRegular',
		fontSize: 17,
		lineHeight: 22.1,
	},
	header: {
		flex: 1,
		marginBottom: 10,
	},
	iconBackground: {
		alignItems: 'center',
		backgroundColor: colors.green,
		borderRadius: 50,
		height: 27,
		justifyContent: 'center',
		width: 27,
	},
	iconContainer: {
		alignSelf: 'flex-start',
		marginRight: 12,
	},
	intStyle: {
		fontFamily: 'MulishRegular',
		fontSize: 22,
		lineHeight: 22.1,
	},
	lastIncome: {
		color: colors.green,
		fontFamily: 'MulishSemiBold',
		fontSize: 14,
	},
	rate: {
		marginLeft: 'auto',
	},
	row: {
		justifyContent: 'space-between',
		width: '100%',
	},
	subheaderText: {
		fontSize: 14,
	},
	topUpButton: {
		backgroundColor: colors.blue,
		flex: 1,
		marginRight: 5,
	},
	withdrawalButton: {
		flex: 1,
		marginLeft: 5,
	},
});
