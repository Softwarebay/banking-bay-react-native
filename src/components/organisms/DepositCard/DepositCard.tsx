import { StyleProp, View, ViewStyle } from 'react-native';

import {
	Button,
	Card,
	Flex,
	FloatNum,
	Heading,
	Icon,
	Text,
} from 'components/atoms';
import { colors } from 'theme';

import { styles } from './DepositCard.styles';

export interface Props {
	isFinished: boolean;
	amount: number;
	currency: string;
	date: string;
	lastIncome: number;
	rate: number;
	testID: string;
	onPressWithdrawalButton: () => void;
	onPressExtendButton: () => void;
	onPressTopUpButton: () => void;
	style?: StyleProp<ViewStyle>;
}

export const DepositCard = ({
	isFinished,
	amount,
	currency,
	date,
	rate,
	lastIncome,
	testID,
	onPressWithdrawalButton,
	onPressExtendButton,
	onPressTopUpButton,
	style,
}: Props) => {
	const onPressSecondaryButton = isFinished
		? onPressExtendButton
		: onPressTopUpButton;

	return (
		<Card direction="column" style={[styles.container, style]} testID={testID}>
			<Flex direction="row">
				<View style={styles.iconContainer}>
					{isFinished ? (
						<View style={styles.iconBackground}>
							<Icon color={colors.white} name="check" size={14} />
						</View>
					) : (
						<Icon color={colors.mainDark} name="deposits" size={27} />
					)}
				</View>
				<Flex direction="column" style={styles.header}>
					<Flex direction="row" style={styles.row}>
						<FloatNum
							floatStyle={styles.floatStyle}
							intStyle={styles.intStyle}
							number={`${(amount / 100).toFixed(2)}`}
							testID="deposit-amount-test-id"
						/>
						<Heading
							size="h5"
							style={styles.currency}
							testID="deposit-currency-test-id"
							text={currency}
						/>
						<Heading
							size="h5"
							style={styles.rate}
							testID="deposit-rate-test-id"
							text={`${(rate * 100).toFixed(0)}%`}
						/>
					</Flex>
					<Flex direction="row" style={styles.row}>
						<Text
							style={styles.subheaderText}
							testID="deposit-date-test-id"
							text={date}
						/>
						<Text
							style={[styles.subheaderText, styles.lastIncome]}
							testID="deposit-last-income-test-id"
							text={`+ ${(lastIncome / 100).toFixed(2)}`}
						/>
					</Flex>
				</Flex>
			</Flex>
			<Flex direction="row">
				<Button
					style={isFinished ? styles.extendButton : styles.topUpButton}
					testID={`deposit-${isFinished ? 'extend' : 'top-up'}-btn-test-id`}
					title={isFinished ? 'Extend' : 'Top - Up'}
					type="ternary"
					onPress={onPressSecondaryButton}
				/>
				<Button
					style={styles.withdrawalButton}
					testID="deposit-withdrawal-btn-test-id"
					title="Withdrawal"
					type="ternary"
					onPress={onPressWithdrawalButton}
				/>
			</Flex>
		</Card>
	);
};
