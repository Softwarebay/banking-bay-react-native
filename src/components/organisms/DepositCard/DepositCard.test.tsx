import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { DepositCard } from './DepositCard';

import type { Props } from './DepositCard';

describe('DepositCard', () => {
	const onPressWithdrawalButton = jest.fn<
		Props['onPressWithdrawalButton'],
		[]
	>();
	const onPressExtendButton = jest.fn<Props['onPressExtendButton'], []>();
	const onPressTopUpButton = jest.fn<Props['onPressTopUpButton'], []>();

	const props: Omit<Props, 'isFinished'> = {
		onPressWithdrawalButton,
		onPressExtendButton,
		onPressTopUpButton,
		testID: 'deposit-card-test-id',
		amount: 100000,
		currency: 'USD',
		date: 'Sep 1 - Mar 1, 2022',
		lastIncome: 6057,
		rate: 0.03,
		style: {
			marginBottom: 10,
		},
	};

	it('should match snapshot - deposit ongoing', () => {
		const tree = testRenderer
			.create(<DepositCard {...props} isFinished={false} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - deposit finished', () => {
		const tree = testRenderer
			.create(<DepositCard {...props} isFinished />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should display current deposit value with currency', () => {
		render(<DepositCard {...props} isFinished />);

		const depositAmount = screen.getByTestId('deposit-amount-test-id');
		const currency = screen.getByTestId('deposit-currency-test-id');

		expect(depositAmount).toHaveTextContent('1000.00');
		expect(currency).toHaveTextContent('USD');
	});

	it('should display percentage of deposit rate', () => {
		render(<DepositCard {...props} isFinished />);

		const rate = screen.getByTestId('deposit-rate-test-id');

		expect(rate).toHaveTextContent('3%');
	});

	it('should display date', () => {
		render(<DepositCard {...props} isFinished />);

		const date = screen.getByTestId('deposit-date-test-id');

		expect(date).toHaveTextContent('Sep 1 - Mar 1, 2022');
	});

	it('should display value of last income', () => {
		render(<DepositCard {...props} isFinished />);

		const lastIncome = screen.getByTestId('deposit-last-income-test-id');

		expect(lastIncome).toHaveTextContent('+ 60.57');
	});

	it('should display two buttons', () => {
		render(<DepositCard {...props} isFinished />);

		const buttons = screen.getAllByTestId(/-btn-test-id/);

		expect(buttons).toHaveLength(2);
	});

	it("should render 'Withdrawal' and 'Extend' buttons for finished deposits", () => {
		render(<DepositCard {...props} isFinished />);

		const withdrawalBtn = screen.getByTestId('deposit-withdrawal-btn-test-id');
		const extendBtn = screen.getByTestId('deposit-extend-btn-test-id');
		const topUpBtn = screen.queryByTestId('deposit-top-up-btn-test-id');

		expect(withdrawalBtn).toHaveTextContent('Withdrawal');
		expect(extendBtn).toHaveTextContent('Extend');
		expect(topUpBtn).toBeNull();
	});

	it('should handle press events on both buttons for finished deposits', () => {
		render(<DepositCard {...props} isFinished />);

		const withdrawalBtn = screen.getByTestId('deposit-withdrawal-btn-test-id');
		const extendBtn = screen.getByTestId('deposit-extend-btn-test-id');

		fireEvent.press(withdrawalBtn);
		expect(onPressWithdrawalButton).toHaveBeenCalledTimes(1);

		fireEvent.press(extendBtn);
		expect(onPressExtendButton).toHaveBeenCalledTimes(1);

		expect(onPressTopUpButton).not.toHaveBeenCalled();
	});

	it("should render 'Withdrawal' and 'Top - Up' buttons for ongoing deposits", () => {
		render(<DepositCard {...props} isFinished={false} />);

		const withdrawalBtn = screen.getByTestId('deposit-withdrawal-btn-test-id');
		const topUpBtn = screen.getByTestId('deposit-top-up-btn-test-id');
		const extendBtn = screen.queryByTestId('deposit-extend-btn-test-id');

		expect(withdrawalBtn).toHaveTextContent('Withdrawal');
		expect(topUpBtn).toHaveTextContent('Top - Up');
		expect(extendBtn).toBeNull();
	});

	it('should handle press events on both buttons for ongoing deposits', () => {
		render(<DepositCard {...props} isFinished={false} />);

		const withdrawalBtn = screen.getByTestId('deposit-withdrawal-btn-test-id');
		const topUpBtn = screen.getByTestId('deposit-top-up-btn-test-id');

		fireEvent.press(withdrawalBtn);
		expect(onPressWithdrawalButton).toHaveBeenCalledTimes(1);

		fireEvent.press(topUpBtn);
		expect(onPressTopUpButton).toHaveBeenCalledTimes(1);

		expect(onPressExtendButton).not.toHaveBeenCalled();
	});

	it('should render check icon for finished deposits', () => {
		render(<DepositCard {...props} isFinished />);

		const checkIcon = screen.getByTestId('check-icon-test-id');
		const depositsIcon = screen.queryByTestId('deposits-icon-test-id');

		expect(checkIcon).not.toBeNull();
		expect(depositsIcon).toBeNull();
	});

	it('should render deposits icon for ongoing deposits', () => {
		render(<DepositCard {...props} isFinished={false} />);

		const depositsIcon = screen.getByTestId('deposits-icon-test-id');
		const checkIcon = screen.queryByTestId('check-icon-test-id');

		expect(depositsIcon).not.toBeNull();
		expect(checkIcon).toBeNull();
	});

	it('should apply additional styles', () => {
		render(<DepositCard {...props} isFinished />);

		const card = screen.getByTestId('deposit-card-test-id');

		expect(card).toHaveStyle({ marginBottom: 10 });
	});
});
