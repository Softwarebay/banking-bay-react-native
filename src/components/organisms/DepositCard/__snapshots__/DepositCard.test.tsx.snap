// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositCard should match snapshot - deposit finished 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "column",
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "shadowColor": "#161966",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        Array [
          Object {
            "padding": 20,
          },
          Object {
            "marginBottom": 10,
          },
        ],
      ],
    ]
  }
  testID="deposit-card-test-id"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "marginRight": 12,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3EB290",
            "borderRadius": 50,
            "height": 27,
            "justifyContent": "center",
            "width": 27,
          }
        }
      >
        <SvgMock
          color="#FFFFFF"
          height={14}
          testID="check-icon-test-id"
          width={14}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "column",
          },
          Object {
            "flex": 1,
            "marginBottom": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
              },
              Object {
                "flexDirection": "row",
              },
              Array [
                Object {
                  "alignItems": "baseline",
                },
                undefined,
              ],
            ]
          }
          testID="deposit-amount-test-id"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1B1D4D",
                },
                Object {
                  "fontFamily": "MulishRegular",
                  "fontSize": 22,
                  "lineHeight": 22.1,
                },
              ]
            }
          >
            1000.
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#1B1D4D",
                },
                Object {
                  "fontFamily": "MulishRegular",
                  "fontSize": 17,
                  "lineHeight": 22.1,
                },
              ]
            }
          >
            00
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Object {
                "fontFamily": "MulishSemiBold",
                "fontSize": 17,
                "lineHeight": 22.1,
              },
              Object {
                "fontFamily": "MulishBold",
                "marginLeft": 5,
              },
            ]
          }
          testID="deposit-currency-test-id"
        >
          USD
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Object {
                "fontFamily": "MulishSemiBold",
                "fontSize": 17,
                "lineHeight": 22.1,
              },
              Object {
                "marginLeft": "auto",
              },
            ]
          }
          testID="deposit-rate-test-id"
        >
          3%
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 17,
                "lineHeight": 27.2,
              },
              Object {
                "fontSize": 14,
              },
            ]
          }
          testID="deposit-date-test-id"
        >
          Sep 1 - Mar 1, 2022
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 17,
                "lineHeight": 27.2,
              },
              Array [
                Object {
                  "fontSize": 14,
                },
                Object {
                  "color": "#3EB290",
                  "fontFamily": "MulishSemiBold",
                  "fontSize": 14,
                },
              ],
            ]
          }
          testID="deposit-last-income-test-id"
        >
          + 60.57
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#3EB290",
            "borderWidth": 0,
            "height": 40,
          },
          Object {
            "backgroundColor": "#FF8A71",
            "flex": 1,
            "marginRight": 5,
          },
        ]
      }
      testID="deposit-extend-btn-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Extend
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#3EB290",
            "borderWidth": 0,
            "height": 40,
          },
          Object {
            "flex": 1,
            "marginLeft": 5,
          },
        ]
      }
      testID="deposit-withdrawal-btn-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Withdrawal
      </Text>
    </View>
  </View>
</View>
`;

exports[`DepositCard should match snapshot - deposit ongoing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "column",
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "shadowColor": "#161966",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        Array [
          Object {
            "padding": 20,
          },
          Object {
            "marginBottom": 10,
          },
        ],
      ],
    ]
  }
  testID="deposit-card-test-id"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "marginRight": 12,
        }
      }
    >
      <SvgMock
        color="#1B1D4D"
        height={27}
        testID="deposits-icon-test-id"
        width={27}
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "column",
          },
          Object {
            "flex": 1,
            "marginBottom": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
              },
              Object {
                "flexDirection": "row",
              },
              Array [
                Object {
                  "alignItems": "baseline",
                },
                undefined,
              ],
            ]
          }
          testID="deposit-amount-test-id"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1B1D4D",
                },
                Object {
                  "fontFamily": "MulishRegular",
                  "fontSize": 22,
                  "lineHeight": 22.1,
                },
              ]
            }
          >
            1000.
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#1B1D4D",
                },
                Object {
                  "fontFamily": "MulishRegular",
                  "fontSize": 17,
                  "lineHeight": 22.1,
                },
              ]
            }
          >
            00
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Object {
                "fontFamily": "MulishSemiBold",
                "fontSize": 17,
                "lineHeight": 22.1,
              },
              Object {
                "fontFamily": "MulishBold",
                "marginLeft": 5,
              },
            ]
          }
          testID="deposit-currency-test-id"
        >
          USD
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Object {
                "fontFamily": "MulishSemiBold",
                "fontSize": 17,
                "lineHeight": 22.1,
              },
              Object {
                "marginLeft": "auto",
              },
            ]
          }
          testID="deposit-rate-test-id"
        >
          3%
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 17,
                "lineHeight": 27.2,
              },
              Object {
                "fontSize": 14,
              },
            ]
          }
          testID="deposit-date-test-id"
        >
          Sep 1 - Mar 1, 2022
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 17,
                "lineHeight": 27.2,
              },
              Array [
                Object {
                  "fontSize": 14,
                },
                Object {
                  "color": "#3EB290",
                  "fontFamily": "MulishSemiBold",
                  "fontSize": 14,
                },
              ],
            ]
          }
          testID="deposit-last-income-test-id"
        >
          + 60.57
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#3EB290",
            "borderWidth": 0,
            "height": 40,
          },
          Object {
            "backgroundColor": "#55ACEE",
            "flex": 1,
            "marginRight": 5,
          },
        ]
      }
      testID="deposit-top-up-btn-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Top - Up
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#3EB290",
            "borderWidth": 0,
            "height": 40,
          },
          Object {
            "flex": 1,
            "marginLeft": 5,
          },
        ]
      }
      testID="deposit-withdrawal-btn-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Withdrawal
      </Text>
    </View>
  </View>
</View>
`;
